src/main/java/net/neoforged/neoforge/data/loading/DatagenModLoader.java

```diff
@@ -55,8 +56,16 @@ public static void begin(final Set<String> mods, final Path path, final Collecti
         }
         setup.run();
         existingFileHelper = new ExistingFileHelper(existingPacks, existingMods, structureValidator, assetIndex, assetsDir);
-        ModLoader.runEventGenerator(mc -> eventGenerator.create(mc, dataGeneratorConfig.makeGenerator(p -> dataGeneratorConfig.isFlat() ? p : p.resolve(mc.getModId()),
-                dataGeneratorConfig.getMods().contains(mc.getModId())), dataGeneratorConfig, existingFileHelper));
+
+        // Only fire the event for mods that have their generators enabled
+        for (ModContainer mod : ModList.get().getSortedMods()) {
```

Comment:
Personally I would have iterated the enabled mods list and posted the event to matching mod containers, rather than iterating the entire `ModList` and checking if a given mod is enabled.

Mainly since theres less to iterate which could make things just a tad faster, but either method works.

Reply:
Iterating over the enabled mods will not respect sorting as is the case with normal events. While it's quite pointless in datagen, I would rather keep it.

Reply:
Thats a good point yeah, better to respect desired mod ordering even if its not strictly needed in datagen. While could be achieved by sorting the enabled mods list or something, that is more work than its worth here.

Reply:
It is useful/potentially necessary for mods like Mekanism which have datagen run for Mekanism and sub mods. While the order of sub mods doesn’t matter, they do have to run after the main Mekanism mod.